app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala

```diff
@@ -226,10 +238,14 @@ class IncomeUpdateCalculatorController @Inject()(
           }
         },
         formData => {
+          journeyCacheService.cache(UpdateIncome_HowToUpdateKey, formData.howToUpdate.getOrElse(""))
```

Comment:
You should map over this. This can cause problems with the future not being executed as you are not waiting for the result to finished before returning from the method.

Something like:
```
journeyCacheService.cache(UpdateIncome_HowToUpdateKey, formData.howToUpdate.getOrElse("")).map { _ =>
  formData.howToUpdate match {
    case Some("incomeCalculator") => Redirect(routes.IncomeUpdateCalculatorController.workingHoursPage())
    case _ => Redirect(controllers.routes.IncomeController.viewIncomeForEdit())
  }
}
```
Would solve the issue

Reply:
Added

Reply:
Not sure what is going on here, but that code is 100% changed to this.

`formData => {
          journeyCacheService.cache(UpdateIncome_HowToUpdateKey, formData.howToUpdate.getOrElse("")).map { _ =>
            formData.howToUpdate match {
              case Some("incomeCalculator") => {
                Redirect(routes.IncomeUpdateCalculatorController.workingHoursPage())
              }
              case _ => {
                Redirect(controllers.routes.IncomeController.viewIncomeForEdit())
              }
            }
          }
        }`